### PR TITLE
Feature/routepage

### PR DESCRIPTION
**Pull request by LeonStechmann**
Very big PR. SORRY in advance, but it just added up so fast in size as the logic got more and more complex.

As it stand, now you are able to plan a bar route based on a start location, a destination , a radius and a amount of stops on the way. 

The onClick "Plan Route" will take you to the route detail page, which I worked on in this issue. 

There you can see the map component with a route, based on user input. A list of bars, that are waypoint on your route, are displayed in green below the map. Additional information about distance and time to next stop/waypoint is also provided below each green box. 

The red boxes indicate the bars, that are not (yet) part of your route, but are also nearby and were extracted by the nearby search function. 

As a user you can add or remove waypoints and bars from your route and it will automatically update the route on the map dynamically. 

Next issue/US will be a much smaller one, probably just styling to what I have so far.

Thanks for taking the time to read through all of this code, I tried my best to keep it readable and somewhat clean, even though further refactoring probably has to be done in the future.

### CODE REVIEW

File: components/Map.js [hunk 1]
```diff
@@ -59,93 +116,159 @@ export default function Map({selected, selectedDest, radius}) {
...
   };
 
+  const logfunction = () => {
+    console.log("bars");
+    console.log(bars);
+    console.log("directionsResponse");
+    console.log(directionsResponse);
+    console.log("waypoints");
+    console.log(waypoints);
+    console.log(directionsResponse.routes[0].legs[0].distance.text);
```
Comment (stefankorder) [inline components/Map.js:189]: You really need this?

File: components/Form.js [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
-      <label htmlFor="stops">
-        <input type="number" name="stops" id="stops" required />
-      </label>
-      <span>Set radius:</span>
-      <label htmlFor="radius">
-        <input type="number" name="radius" id="radius" required />
-      </label>
-      <StyledButton type="submit">Submit</StyledButton>
-    </FormContainer>
+    <>
```
Comment (stefankorder) [inline components/Form.js:22]: I think you don't need a Fragment at this point.

**Review by stefankorder — CHANGES_REQUESTED** [review]
LGTM! Just 1 hint and 1 question.

File: components/Map.js (hunk 1 above)
Comment (LeonStechmann) [reply components/Map.js:189]: Woops. No, I just had this to check my various states. 
It was super useful for debugging and refactoring. I will remove it

File: components/Form.js (hunk 2 above)
Comment (LeonStechmann) [reply components/Form.js:22]: True! I removed it.

**Pull request merged**